File: runme/mcgen/local/datan/exasrc/e2av.c
#include <stdio.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__4 = 4, c__6 = 6;

main(void)
{
    static integer i, j, k, ntype, iseed1, iseed2, ndf[6];
    static double xbij[16] /* was [4][4] */, f[4], q[6], s[6], alpha[4],
                  x[96] /* was [4][4][6] */, sigma, xb, buffer[6],
                  xbi[4], xbj[4];

/* identify program to user */
    printf("%s\n", "program E2AV generates data and performs");
    printf("%s\n\n", "analysis of variance on them.");
/* ask for input */
    printf("%s\n", "Enter SIGMA (>0.):");
    printf("%s", "> ");
    scanf("%lf", &sigma);
    printf("%s\n", "Enter ISEED1 (>0):");
    printf("%s", "> ");
    scanf("%li", &iseed1);
    printf("%s\n", "Enter ISEED2 (>0):");
    printf("%s", "> ");
    scanf("%li", &iseed2);
    printf("%s\n", "Choose:");
    printf("%s\n", "0 - crossed classification");
    printf("%s\n", "1 - nested classification");
    printf("%s", "> ");
    scanf("%li", &ntype);
/* initialize random number generator */
    rne2in_(&iseed1, &iseed2);
/* generate data */
    for (i = 1; i <= 4; ++i) {
	for (j = 1; j <= 4; ++j) {
	    rnstnr_(buffer, &c__6);
	    for (k = 1; k <= 6; ++k) {
		x[i + (j + (k << 2) << 2) - 21] = buffer[k - 1] * sigma;
	    }
	}
    }
    avtble_(x, &c__4, &c__4, &c__6, &xb, xbi, xbj, xbij, q, s, f, ndf, alpha);
    avoutp_(x, &c__4, &c__4, &c__6, q, s, f, ndf, alpha, &ntype);
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e2av.f
      PROGRAM E2AV
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      PARAMETER(NI=4,NJ=4,NK=6)
      DIMENSION BUFFER(NK)
      DIMENSION X(NI,NJ,NK),XBI(NI),XBJ(NJ),XBIJ(NI,NJ)
      DIMENSION Q(6),S(6),F(4),NDF(6),ALPHA(4)
C identify program to user
      WRITE(*,*)' program E2AV generates data and performs '
      WRITE(*,*)' analysis of variance on them '
      WRITE(*,*)' '
C ask for input
      WRITE(*,*)' Enter SIGMA (>0.)'
      WRITE(*,*)' > '
      READ(*,*)SIGMA
      WRITE(*,*)' Enter ISEED1 (>0)'
      WRITE(*,*)' > '
      READ(*,*)ISEED1
      WRITE(*,*)' Enter ISEED2 (>0)'
      WRITE(*,*)' > '
      READ(*,*)ISEED2
      WRITE(*,*)' Choose '
      WRITE(*,*)' 0 - crossed classification '
      WRITE(*,*)' 1 - nested classification '
      WRITE(*,*)' > '
      READ(*,*)NTYPE
C initialize random number generator
      CALL RNE2IN(ISEED1,ISEED2)
C generate data
      DO 30 I=1,NI
        DO 20 J=1,NJ
          CALL RNSTNR(BUFFER,NK)
          DO 10 K=1,NK
            X(I,J,K)=BUFFER(K)*SIGMA
10        CONTINUE
20      CONTINUE
30    CONTINUE
      CALL AVTBLE(X,NI,NJ,NK,XB,XBI,XBJ,XBIJ,Q,S,F,NDF,ALPHA)
      CALL AVOUTP(X,NI,NJ,NK,Q,S,F,NDF,ALPHA,NTYPE)
      END
